--- a/setup-storage	2019-03-27 22:44:02.000000000 +0100
+++ b/setup-storage	2020-09-10 23:51:05.923093090 +0200
@@ -261,9 +261,22 @@
   foreach (keys %FAI::disk_var);
 
 if ($FAI::no_dry_run) {
+  # List all bootable devices. Can be used by later tasks to install a
+  # boot loader. Using Variable PHYSICAL_BOOT_DEVICES.
+  my @boot_devices;
+  foreach (keys %FAI::configs) {
+    $_ =~ /PHY_(.+)/;
+    if ( $1 && defined $FAI::configs{$_}{'bootable'}
+      &&  $FAI::configs{$_}{'bootable'} > 0 )  {
+      push @boot_devices, $1;
+    }
+  }
+  @boot_devices = sort @boot_devices;
+
   open(DISK_VAR, ">$FAI::DATADIR/disk_var.sh")
     or die "Unable to write to file $FAI::DATADIR/disk_var.sh\n";
   print DISK_VAR "$_=\${$_:-$FAI::disk_var{$_}}\n" foreach (keys %FAI::disk_var);
+  print DISK_VAR "PHYSICAL_BOOT_DEVICES=\"", join(" ", @boot_devices), "\"\n";
   close DISK_VAR;
 
   if ($opt_y) {
@@ -282,6 +295,9 @@
 	print DISK_YML "$_: $FAI::disk_var{$_}\n";
       }
     }
+    print DISK_YML "PHYSICAL_BOOT_DEVICES:\n";
+    my $a = join("\n  - ", @boot_devices);
+    print DISK_YML "  - $a\n";
     close DISK_YML;
   }
 }
